modules/raw-rsa-keyring-node/test/raw_rsa_keyring_node.test.ts

```diff
@@ -0,0 +1,182 @@
+
+describe('RawRsaKeyringWebCrypto::constructor', () => {
+  const keyName = 'keyName'
+  const keyNamespace = 'keyNamespace'
+
+  it('constructor decorates', async () => {
+    const test = new RawRsaKeyringNode({
+      rsaKey: {
+        privateKey: privatePem,
+        publicKey: publicPem
+      },
+      keyName,
+      keyNamespace
+    })
+
+    expect(test.keyName).to.equal(keyName)
+    expect(test.keyNamespace).to.equal(keyNamespace)
+    expect(test._wrapKey).to.be.a('function')
+    expect(test._unwrapKey).to.be.a('function')
+    expect(test).to.be.instanceOf(KeyringNode)
+  })
+
+  it('can construct with only public key', () => {
+    const testPublicOnly = new RawRsaKeyringNode({
+      rsaKey: {
+        publicKey: publicPem
+      },
+      keyName,
+      keyNamespace
+    })
+    expect(testPublicOnly).to.be.instanceOf(RawRsaKeyringNode)
+  })
+
+  it('can construct with only private key', () => {
+    const testPrivateOnly = new RawRsaKeyringNode({
+      rsaKey: {
+        privateKey: privatePem
+      },
+      keyName,
+      keyNamespace
+    })
+    expect(testPrivateOnly).to.be.instanceOf(RawRsaKeyringNode)
+  })
+
+  it('Precondition: RsaKeyringNode needs either a public or a private key to operate.', () => {
+    expect(() => new RawRsaKeyringNode({
+      keyName,
+      keyNamespace,
+      rsaKey: {}
+    })).to.throw()
+  })
+
+  it('Precondition: RsaKeyringNode needs identifying information for encrypt and decrypt.', () => {
+    // @ts-ignore Typescript is trying to save us.
```

Comment:
What is typescript saving us from?

Reply:
Typescript knows that this function should be called with a `keyNamespace` and a `keyName`, so it will complain.  But JS does not have this protection... So to test that this will indeed throw, I need TypeScript to be quite and not complain about calling the function incorrectly.

Reply:
👍